x-pack/plugins/ai_infra/product_doc_base/server/services/search/perform_search.ts

```diff
@@ -5,29 +5,28 @@
  * 2.0.
  */
 
-import type { Client } from '@elastic/elasticsearch';
+import type { ElasticsearchClient } from '@kbn/core/server';
+import type { ProductDocumentationAttributes } from '@kbn/product-doc-common';
 
 // https://search-labs.elastic.co/search-labs/blog/elser-rag-search-for-relevance
 
-export const performSemanticSearch = async ({
+export const performSearch = async ({
   searchQuery,
+  size,
   index,
   client,
 }: {
   searchQuery: string;
-  index: string;
-  client: Client;
+  size: number;
+  index: string | string[];
```

Comment:
The product documentation search query. For now, I just reused the exact query (adapted for semantic_text) that the innovation team is using for their own documentation retrieval. I assumed this made sense given we're using the exact same content anyway. But there could be areas of improvements in that search query (rerankers/non-standard-retriever for example).